src/ds5/ds5-device.cpp

```diff
@@ -350,7 +350,7 @@ namespace librealsense
 
             // Activate Thermal Compensation tracking
             if (supports_option(RS2_OPTION_THERMAL_COMPENSATION) && (get_option(RS2_OPTION_THERMAL_COMPENSATION).query()>0.f))
-                this->_owner->_thermal_monitor->update(true);
+                _owner->_thermal_monitor->update(true);
```

Comment:
Shouldn't you `update` with `get_option(RS2_OPTION_THERMAL_COMPENSATION).query()>0.f`?
I.e.:
```
if( supports_option(...) )
    _owner->_thermal_monitor->update( get_option(RS2_OPTION_THERMAL_COMPENSATION).query()>0.f );
```
?